SimG4Core/Notification/src/GenParticleInfoExtractor.cc

```diff
@@ -1,14 +1,18 @@
 #include "SimG4Core/Notification/interface/GenParticleInfoExtractor.h"
-#include "SimG4Core/Notification/interface/SimG4Exception.h"
-
 #include "G4PrimaryParticle.hh"
 
 const GenParticleInfo &GenParticleInfoExtractor::operator()(const G4PrimaryParticle *p) const {
   G4VUserPrimaryParticleInformation *up = p->GetUserInformation();
   if (up == nullptr)
-    throw SimG4Exception("GenParticleInfoExtractor: G4PrimaryParticle has no user information");
+    G4Exception("SimG4Core/Notification",
```

Comment:
Uhm, I was too fast in signing this before inspecting the static analyzer report.

In the previous code if `up == nullptr` we were throwing here. Now, there is this G4Exception (is it also throwing?), and therefore a null reference can possibly be returned by this method. Is it this the case? Is it the expected/wanted behaviour?